### PR TITLE
3: GET /api/topics complete

### PR DESCRIPTION
**Pull request by dan77uk**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const express = require("express");
+const app = express();
+const { getTopics } = require("./controllers/controllers");
+
+app.use(express.json());
```
Comment (emilyb93) [inline app.js]: We dont have any end points that require a body yet, this should not be added until it needs to be used

File: __tests__/getTopics.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        expect(body.topics).toHaveLength(3);
+        body.topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            slug: expect.any(String),
+            description: expect.any(String),
+          });
+        });
+      });
+  });
+});
```
Comment (emilyb93) [inline __tests__/getTopics.test.js:38]: I would like to see a test for a broken path such as '/api/topicz'

File: app.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+app.get("/api/topics", getTopics);
+app.use((err, req, res, next) => {
+  res.status(err.status).send({ msg: err.msg });
+});
+
```
Comment (emilyb93) [inline app.js:12]: Would like to see an explicit 404 error handler in your app file

**Review by emilyb93 — CHANGES_REQUESTED** [review]
We need to explicitly handle and test for a 404 error to prove that we will get one when the client request is incorrect

**Review by hevdevs — COMMENTED** [review]
Looks good to merge!

**Pull request merged**